server/src/main/java/io/crate/execution/engine/collect/DocValuesGroupByOptimizedIterator.java

```diff
@@ -98,24 +97,23 @@
             return null;
         }
 
-        ArrayList<Reference> docKeyRefs = new ArrayList<>(groupProjection.keys().size());
-        for (var key : groupProjection.keys()) {
-            var keyRef = getKeyRef(collectPhase.toCollect(), key);
-            if (keyRef == null) {
-                return null; // group by on non-reference
-            } else {
-                docKeyRefs.add(keyRef);
-            }
+        if (Symbols.containsColumn(collectPhase.toCollect(), DocSysColumns.SCORE)
```

Comment:
why not as the very first check even before checking for shard projections?

Reply:
ah true, changed it and squashed the fixup